river/Seat.zig

```diff
@@ -230,11 +233,34 @@ pub fn setFocusRaw(self: *Self, new_focus: FocusTarget) void {
         .none => null,
     };
 
+    // Keep the parent top-level Xwayland view of any override redirect surface
+    // activated while that override redirect surface is focused. This ensures
+    // override redirect menus do not disappear as a result of deactivating
+    // their parent window.
+    if (new_focus == .xwayland_override_redirect and
+        self.focused == .view and
+        self.focused.view.impl == .xwayland_view and
+        self.focused.view.impl.xwayland_view.xwayland_surface.pid == new_focus.xwayland_override_redirect.xwayland_surface.pid)
+    {
+        self.activated_xwayland_view = self.focused.view;
+    } else if (self.activated_xwayland_view) |active_view| {
+        if (!(new_focus == .view and new_focus.view == active_view) and
+            !(new_focus == .xwayland_override_redirect and new_focus.xwayland_override_redirect.xwayland_surface.pid == active_view.impl.xwayland_view.xwayland_surface.pid))
+        {
+            if (active_view.pending.focus == 0) active_view.setActivated(false);
+            self.activated_xwayland_view = null;
+        }
+    }
+
     // First clear the current focus
     switch (self.focused) {
         .view => |view| {
             view.pending.focus -= 1;
-            if (view.pending.focus == 0) view.setActivated(false);
+            if (view != self.activated_xwayland_view and
+                view.pending.focus == 0)
+            {
+                view.setActivated(false);
+            }
```

Comment:
We'll need to wrap this with a check for the build_option to keep this compiling when xwayland is disabled.
```suggestion
            if (view.pending.focus == 0 and
                (!build_options.xwayland or view != self.activated_xwayland_view)
            {
                view.setActivated(false);
            }
```